client/src/components/UserSettings/SettingsTagColor.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Collapse, Tag } from 'antd';
+import { BgColorsOutlined } from '@ant-design/icons';
+import { GithubPicker } from 'react-color';
+import { pickerColors, mockedTags as tags, setTagColor, getTagColor, DEFAULT_COLOR } from './userSettingsHandlers';
+import { useLocalStorage } from 'react-use';
+
+const SettingsTagColor: React.FC = () => {
+  const { Panel } = Collapse;
+  const defaultTagColor = JSON.stringify({ default: DEFAULT_COLOR });
+  const [storedTagColors, setStoredTagColors] = useLocalStorage<string>('tagColors', defaultTagColor);
+
+  const collapseTags = (
+    <Collapse accordion ghost>
+      {tags.map((item) => {
+        return (
+          <Panel
+            header={
+              <Tag
+                style={{
+                  cursor: 'pointer',
+                  borderColor: getTagColor(item.name, storedTagColors),
+                  color: getTagColor(item.name, storedTagColors),
+                  backgroundColor: `${getTagColor(item.name, storedTagColors)}10`,
+                }}
+              >
+                {item.name}
+              </Tag>
+            }
+            key={item.name}
+          >
+            <GithubPicker
+              colors={pickerColors}
+              triangle="hide"
+              width={'138px'}
+              onChange={(e) => setTagColor(e, item.name, setStoredTagColors, storedTagColors)}
```

Comment:
Let's consider using `useCallback` for onChange handler

Reply:
https://www.codementor.io/blog/react-optimization-5wiwjnf9hj#7-avoid-inline-function-definition-in-the-render-function